Lib/test/test_signal.py

```diff
@@ -423,6 +423,74 @@ def handler(signum, frame):
         """.format(action=action)
         assert_python_ok('-c', code)
 
+    @unittest.skipIf(_testcapi is None, 'need _testcapi')
+    def test_warn_on_full_buffer(self):
+        # Use a subprocess to have only one thread.
+        if os.name == 'nt':
+            action = 'send'
+        else:
+            action = 'write'
+        code = """if 1:
+        import errno
+        import signal
+        import socket
+        import sys
+        import time
+        import _testcapi
+        from test.support import captured_stderr
+
+        signum = signal.SIGINT
+
+        def handler(signum, frame):
+            pass
+
+        signal.signal(signum, handler)
+
+        read, write = socket.socketpair()
+        read.setblocking(False)
+        write.setblocking(False)
+
+        # Fill the send buffer
+        try:
+            while True:
+                write.send(b"x" * 1024)
```

Comment:
What is send(1024 bytes) fails with BlockingIOError, but send(1 bytes) works?

I suggest to loop on write.send(b"x") instead.

Reply:
I don't think this can happen in any real system (you'd get a partial `send` instead), but I checked and sending 1 byte at a time is only ~2x slower, so why not. Done.